File: daq/slc/dqm/src/Graph1F.cc
#include "daq/slc/dqm/Graph1F.h"

#include "daq/slc/dqm/FloatArray.h"

using namespace Belle2;

Graph1F::Graph1F() : Graph1()
{
  m_data = new FloatArray(0);
}

Graph1F::Graph1F(const Graph1F& h) : Graph1(h)
{
  m_data = new FloatArray(*h.m_data);
}

Graph1F::Graph1F(const std::string& name, const std::string& title,
                 int nbinx, double xmin, double xmax, double ymin, double ymax)
  : Graph1(name, title, nbinx, xmin, xmax, ymin, ymax)
{
  m_data = new FloatArray(nbinx * 2);
}

Graph1F::~Graph1F() throw() {}

std::string Graph1F::getDataType() const throw()
{
  return std::string("g1F");
}

File: daq/slc/extra/javalib/b2daq/src/b2daq/dqm/core/Graph1F.java
package b2daq.dqm.core;


public class Graph1F extends Graph1 {

	public Graph1F() {
		super();
		data = new FloatArray(0);
	}
	public Graph1F(String name, String title, int nbinx, double xmin, double xmax,
			double ymin, double ymax) {
		super(name, title, nbinx, xmin, xmax, ymin, ymax);
		data = new FloatArray(nbinx*2);
	}
	public Graph1F(Graph1F h) {
		super();
		data = new FloatArray(0);
	}
	public String getDataType() {
		return "g1F";
	}

}
